include/deal.II/base/symmetric_tensor.h

```diff
@@ -985,14 +996,27 @@ SymmetricTensor<rank,dim,Number>::SymmetricTensor (const Number (&array) [n_inde
 
 
 template <int rank, int dim, typename Number>
+template <typename OtherNumber>
+inline
+SymmetricTensor<rank,dim,Number> &
+SymmetricTensor<rank,dim,Number>::operator = (const SymmetricTensor<rank,dim,OtherNumber> &t)
+{
+  for (unsigned int i=0; i<base_tensor_type::dimension; ++i)
+    data[i] = t.data[i];
```

Comment:
same question here about what happens for rank-4 tensors with AD types or complex types